### ISSUE TITLE
Errors in modules are not being emitted correctly

### ISSUE THREAD

**Issue by njunxuan** [wata727/tflint#493]
Hi, thank you for this wonderful tool. I am currently integrating tflint into my setup and noticed that lint errors for the `terraform_dash_in_resource_name` do not show up for resources contained in modules (I have enabled the rule using the command line options).

I have traced through the code and I believe the reason for this is that inside `EmitIssue`, the `else` portion of the code is being triggered and `r.listModuleVars` is an empty list. Thus, the issues do not get added to `r.Issues`. Removing the `if` condition fixes my issue and produces all the expected issues.

I am not sure why there is a loop in the `else` portion and there is no documentation for the `listModuleVars` function, so I was hoping you could help with explaining the purpose of the loop. Thanks!

An example:
```hcl
# main.tf
module "my_module" {
   source = "../modules/example_module"
}

# ../modules/example_module
resource "aws_route53_resolver_endpoint" "test-resolver" { <-- this should raise a lint error
  # ... 
}
```

## Version

TFLint v0.12.1
Terraform v0.12.9

**Comment by njunxuan** [comment]
Thanks for getting back. Looking at your PR, tflint should still throw an error for my example right, just at that root module level? Or is it necessary for me to run tflint in each module directory to detect resources which violate the `terraform_dash_in_resource_name` rule.

**Comment by wata727** [comment]
In your example, the `test-resolver`'s issue is unrelated to the `my_module` module argument declared in `main.tf`. In other words, no matter how you declare the module in `main.tf`, this issue cannot be avoided. TFLint doesn't recognize these issues as `main.tf` issues.

Since this is a issue of `../modules/example_module` itself, I think that in order to detect this, we should run a check on each module directory.

**Comment by njunxuan** [comment]
Thanks for the explanation! Guess I will setup my scripts to recursively check the directories.
